zio-kafka/src/main/scala/zio/kafka/consumer/internal/RunloopAccess.scala

```diff
@@ -88,15 +89,25 @@ private[consumer] object RunloopAccess {
                         hasGroupId = settings.hasGroupId,
                         consumer = consumerAccess,
                         pollTimeout = settings.pollTimeout,
+                        maxPollInterval = maxPollIntervalConfig(settings),
                         diagnostics = diagnostics,
                         offsetRetrieval = settings.offsetRetrieval,
                         userRebalanceListener = settings.rebalanceListener,
                         restartStreamsOnRebalancing = settings.restartStreamOnRebalancing,
                         partitionsHub = partitionsHub,
-                        runloopTimeout = settings.runloopTimeout,
                         fetchStrategy = settings.fetchStrategy
                       )
                       .withFinalizer(_ => runloopStateRef.set(RunloopState.Finalized))
                       .provide(ZLayer.succeed(consumerScope))
     } yield new RunloopAccess(runloopStateRef, partitionsHub, makeRunloop, diagnostics)
+
+  private def maxPollIntervalConfig(settings: ConsumerSettings): Duration = {
+    def defaultMaxPollInterval: AnyRef = ConsumerConfig
+      .configDef()
+      .defaultValues()
+      .getOrDefault(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, "300000")
+    val maxPollIntervalStr = settings.properties
+      .getOrElse(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, defaultMaxPollInterval)
+    maxPollIntervalStr.toString.toInt.millis
+  }
```

Comment:
To me, this isn't safe enough. The `toInt` can fail too.
Here's what I'd like to suggest:
```suggestion
  private def maxPollIntervalConfig(settings: ConsumerSettings): Duration = {
    // 5 minutes.
    // See https://docs.confluent.io/platform/current/installation/configuration/consumer-configs.html#max-poll-interval-ms
    def kafkaDefaultValue = 5.minutes

    def defaultMaxPollInterval: AnyRef =
      ConsumerConfig
        .configDef()
        .defaultValues()
        .getOrDefault(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, kafkaDefaultValue.toMillis)

    try
      settings.properties
        .getOrElse(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, defaultMaxPollInterval)
        .toString
        .toInt
        .millis
    catch {
      case NonFatal(_) => kafkaDefaultValue
    }
  }
```

Reply:
Fixed, but a little bit differently in https://github.com/zio/zio-kafka/pull/960/commits/6830e938d49884e722cb15fdfe5b61775e6b9dfe